lib/ansible/plugins/callback/foreman.py

```diff
@@ -80,154 +80,173 @@
+    """
+    return datetime.utcnow().isoformat()
+
+
 class CallbackModule(CallbackBase):
     CALLBACK_VERSION = 2.0
     CALLBACK_TYPE = 'notification'
     CALLBACK_NAME = 'foreman'
     CALLBACK_NEEDS_WHITELIST = True
 
-    FOREMAN_HEADERS = {
-        "Content-Type": "application/json",
-        "Accept": "application/json"
-    }
-    TIME_FORMAT = "%Y-%m-%d %H:%M:%S %f"
-
     def __init__(self):
         super(CallbackModule, self).__init__()
         self.items = defaultdict(list)
-        self.start_time = int(time.time())
+        self.facts = defaultdict(dict)
+        self.start_time = get_time()
 
     def set_options(self, task_keys=None, var_options=None, direct=None):
 
         super(CallbackModule, self).set_options(task_keys=task_keys, var_options=var_options, direct=direct)
 
-        self.FOREMAN_URL = self.get_option('url')
-        self.FOREMAN_SSL_CERT = (self.get_option('client_cert'), self.get_option('client_key'))
-        self.FOREMAN_SSL_VERIFY = str(self.get_option('verify_certs'))
-
-        self.ssl_verify = self._ssl_verify()
+        self.foreman_url = self.get_option('url')
+        ssl_cert = self.get_option('client_cert')
+        ssl_key = self.get_option('client_key')
 
         if HAS_REQUESTS:
-            requests_major = int(requests.__version__.split('.')[0])
-            if requests_major < 2:
-                self._disable_plugin('The `requests` python module is too old.')
+            requests_version = requests.__version__.split('.')
+            if int(requests_version[0]) < 2 or int(requests_version[1]) < 14:
+                self._disable_plugin(u'The `requests` python module is too old.')
         else:
-            self._disable_plugin('The `requests` python module is not installed.')
+            self._disable_plugin(u'The `requests` python module is not installed.')
 
-        if self.FOREMAN_URL.startswith('https://'):
-            if not os.path.exists(self.FOREMAN_SSL_CERT[0]):
-                self._disable_plugin('FOREMAN_SSL_CERT %s not found.' % self.FOREMAN_SSL_CERT[0])
+        if self.foreman_url.startswith('https://'):
+            if not os.path.exists(ssl_cert):
+                self._disable_plugin(u'FOREMAN_SSL_CERT %s not found.' % ssl_cert)
 
-            if not os.path.exists(self.FOREMAN_SSL_CERT[1]):
-                self._disable_plugin('FOREMAN_SSL_KEY %s not found.' % self.FOREMAN_SSL_CERT[1])
+            if not os.path.exists(ssl_key):
+                self._disable_plugin(u'FOREMAN_SSL_KEY %s not found.' % ssl_key)
+
+        self.session = requests.Session()
+        self.session.verify = self._ssl_verify(str(self.get_option('verify_certs')))
+        self.session.cert = (ssl_cert, ssl_key)
 
     def _disable_plugin(self, msg):
         self.disabled = True
         if msg:
-            self._display.warning(msg + ' Disabling the Foreman callback plugin.')
+            self._display.warning(msg + u' Disabling the Foreman callback plugin.')
```

Comment:
We should probably explicitly `to_text` it
```suggestion
            self._display.warning(to_text(msg) + u' Disabling the Foreman callback plugin.')
```